### ISSUE TITLE
Unable to run flutter 2 in mac desktop for completely new project

### ISSUE THREAD

**Issue by spike04** [flutter/flutter#77325]
Hi there, After upgrading flutter from 1.22.6(stable) to flutter 2.0.1 (stable). I am not able to run flutter for desktop. I can successfully run the application in ios and android. 

ERROR Log while trying to run app for macOS desktop:
```
App.framework/Versions/A/Resources/flutter_assets/vm_snapshot_data
App.framework/Versions/A/Resources/flutter_assets/fonts/
App.framework/Versions/A/Resources/flutter_assets/fonts/MaterialIcons-Regular.otf
App.framework/Versions/A/Resources/flutter_assets/packages/
App.framework/Versions/A/Resources/flutter_assets/packages/cupertino_icons/
App.framework/Versions/A/Resources/flutter_assets/packages/cupertino_icons/assets/
App.framework/Versions/A/Resources/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf

sent 39937381 bytes  received 312 bytes  79875386.00 bytes/sec
total size is 39931395  speedup is 1.00
building file list ... done
FlutterMacOS.framework/
FlutterMacOS.framework/FlutterMacOS
FlutterMacOS.framework/Resources/
FlutterMacOS.framework/Resources/Info.plist
FlutterMacOS.framework/Resources/icudtl.dat
FlutterMacOS.framework/Versions/
FlutterMacOS.framework/Versions/A/
FlutterMacOS.framework/Versions/A/FlutterMacOS
FlutterMacOS.framework/Versions/A/Resources/
FlutterMacOS.framework/Versions/A/Resources/Info.plist
FlutterMacOS.framework/Versions/A/Resources/icudtl.dat
FlutterMacOS.framework/Versions/A/_CodeSignature/
FlutterMacOS.framework/Versions/A/_CodeSignature/CodeResources
FlutterMacOS.framework/Versions/Current/
FlutterMacOS.framework/Versions/Current/FlutterMacOS
FlutterMacOS.framework/Versions/Current/Resources/
FlutterMacOS.framework/Versions/Current/Resources/Info.plist
FlutterMacOS.framework/Versions/Current/Resources/icudtl.dat
FlutterMacOS.framework/Versions/Current/_CodeSignature/
FlutterMacOS.framework/Versions/Current/_CodeSignature/CodeResources

sent 120755397 bytes  received 316 bytes  80503808.67 bytes/sec
total size is 120739515  speedup is 1.00
... build/macos/Build/Products/Debug/prc_demo.app/Contents/Frameworks/App.framework/App: signed bundle with Mach-O thin (x86_64) [io.flutter.flutter.app]
... build/macos/Build/Products/Debug/prc_demo.app/Contents/Frameworks/FlutterMacOS.framework/FlutterMacOS: replacing existing signature
... build/macos/Build/Products/Debug/prc_demo.app/Contents/Frameworks/FlutterMacOS.framework/FlutterMacOS: bundle format is ambiguous (could be app or framework)
Command PhaseScriptExecution failed with a nonzero exit code
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description
** BUILD FAILED **

Exception: Build process failed

```

**Comment by lordgreg** [comment]
Adding the `-v` part where the exception occurs:
```
/Users/foo/Projects//flutter2/build/macos/Build/Products/Release/flutter2.app/Contents/Info.plist
[   +2 ms] GenerateDSYMFile /Users/foo/Projects//flutter2/build/macos/Build/Products/Release/flutter2.app.dSYM
/Users/foo/Projects//flutter2/build/macos/Build/Products/Release/flutter2.app/Contents/MacOS/flutter2 (in target 'Runner' from project 'Runner')
[        ]     cd /Users/foo/Projects//flutter2/macos
[        ]     /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/dsymutil
/Users/foo/Projects//flutter2/build/macos/Build/Products/Release/flutter2.app/Contents/MacOS/flutter2 -o
/Users/foo/Projects//flutter2/build/macos/Build/Products/Release/flutter2.app.dSYM
[   +6 ms] ** BUILD FAILED **
[  +16 ms] The following build commands failed:
[        ] 	PhaseScriptExecution Run\ Script /Users/foo/Projects//flutter2/build/macos/Build/Intermediates.noindex/Runner.build/Release/Runner.build/Script-3399D490228B24CF009A79C7.sh
[        ] (1 failure)
[  +39 ms] Building macOS application... (completed in 6.8s)
[   +6 ms] "flutter macos" took 11,172ms.
[  +23 ms] Build process failed
[   +1 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      buildMacOS (package:flutter_tools/src/macos/build_macos.dart:122:5)
           <asynchronous suspension>
           #2      BuildMacosCommand.runCommand (package:flutter_tools/src/commands/build_macos.dart:51:5)
           <asynchronous suspension>
           #3      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1157:12)
           <asynchronous suspension>
           #4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1009:27)
           <asynchronous suspension>
           #5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #6      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #7      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
           <asynchronous suspension>
           #8      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:278:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #11     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:234:5)
           <asynchronous suspension>
           #12     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:64:9)
           <asynchronous suspension>
           #13     run.<anonymous closure> (package:flutter_tools/runner.dart:62:12)
           <asynchronous suspension>
           #14     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #15     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #16     runInContext (package:flutter_tools/src/context_runner.dart:73:10)
           <asynchronous suspension>
           #17     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>
           
           
[ +235 ms] ensureAnalyticsSent: 232ms
[   +2 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```

**Comment by TahaTesser** [comment]
Hi @spike04 
Since you upgrade from the previous `stable` which had no support, did you generate mac with `flutter create .` and then `flutter run -d macos`?


I just a created project on `1.22.6` and then ran it on `2.0.1`, with `flutter create .` and `flutter run -d maocs` and app lunches fine with no errors and `flutter build macos` works fine too

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 1.22.6 at /Users/tahatesser/Code/flutter
    • Framework revision 9b2d32b605 (6 weeks ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.4)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.
```

</details>

```
tahatesser@Tahas-MacBook-Pro AndroidStudioProjects % cd mydemo 
tahatesser@Tahas-MacBook-Pro mydemo % ls
README.md	ios		mydemo.iml	pubspec.yaml
android		lib		pubspec.lock	test
tahatesser@Tahas-MacBook-Pro mydemo % 
```

```bash
tahatesser@Tahas-MacBook-Pro mydemo % flutters create .    
Recreating project ....
  web/favicon.png (created)
  web/index.html (created)
  web/manifest.json (created)
  web/icons/Icon-192.png (created)
  web/icons/Icon-512.png (created)
  macos/Runner.xcworkspace/contents.xcworkspacedata (created)
  macos/Runner.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (created)
  macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_16.png (created)
  macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_1024.png (created)
  macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_256.png (created)
  macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_64.png (created)
  macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_512.png (created)
  macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_128.png (created)
  macos/Runner/Assets.xcassets/AppIcon.appiconset/Contents.json (created)
  macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_32.png (created)
  macos/Runner/DebugProfile.entitlements (created)
  macos/Runner/Base.lproj/MainMenu.xib (created)
  macos/Runner/MainFlutterWindow.swift (created)
  macos/Runner/Configs/Debug.xcconfig (created)
  macos/Runner/Configs/Release.xcconfig (created)
  macos/Runner/Configs/Warnings.xcconfig (created)
  macos/Runner/Configs/AppInfo.xcconfig (created)
  macos/Runner/AppDelegate.swift (created)
  macos/Runner/Info.plist (created)
  macos/Runner/Release.entitlements (created)
  macos/Runner.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist (created)
  macos/Runner.xcodeproj/project.pbxproj (created)
  macos/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme (created)
  macos/Flutter/Flutter-Debug.xcconfig (created)
  macos/Flutter/Flutter-Release.xcconfig (created)
  macos/.gitignore (created)
  android/app/src/main/res/values-night/styles.xml (created)
  android/app/src/main/res/drawable-v21/launch_background.xml (created)
Running "flutter pub get" in mydemo...                             539ms
Wrote 36 files.

All done!
In order to run your application, type:

  $ cd .
  $ flutter run

To enable null safety, type:

  $ cd .
  $ dart migrate --apply-changes

Your application code is in ./lib/main.dart.

tahatesser@Tahas-MacBook-Pro mydemo % flutters run -d macos
Launching lib/main.dart on macOS in debug mode...
Building macOS application...                                           
Activating Dart DevTools...                                         5.9s
Syncing files to device macOS...                                    69ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h Repeat this help message.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).
An Observatory debugger and profiler on macOS is available at: http://127.0.0.1:60563/w6qmJ8B-sIA=/

Flutter DevTools, a Flutter debugger and profiler, on macOS is available at:
http://127.0.0.1:9100?uri=http%3A%2F%2F127.0.0.1%3A60563%2Fw6qmJ8B-sIA%3D%2F

Running with unsound null safety
For more information see https://dart.dev/null-safety/unsound-null-safety
```

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 2.0.1, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 2.0.1 at /Users/tahatesser/Code/flutter_stable
    • Framework revision c5a4b4029c (17 hours ago), 2021-03-04 09:47:48 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 11.2.2 20D80 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 88.0.4324.192

• No issues found!
```

</details>


Can you please provide reproducible steps and a code sample if needed to reproduce this?
Thank you

**Comment by spike04** [comment]
These are the steps I did:

1. downloaded zipped stable 2.0.0 flutter sdk 
2. Created New Project from VSCode.
3. Then without touching any file just ran the project in debug mode then i get the error message as mentioned above. 

Don't know what's the issue. Since I created completely new file i assumed it should run normally.

**Comment by lordgreg** [comment]
Same here again,

1. no flutter sdk installed before 
2. enabled macos with `flutter config --enable-macos-desktop`
2. downloaded and installed zip from flutter.dev (2.0.0)
3. used CLI to create completely fresh project with `flutter create foo` (tested existing project where I removed the `macos` directory beforehand completely)
4. I've started the build process with `flutter build macos`
5. Tried `flutter -d run macos` - same issue

**Comment by TahaTesser** [comment]
Hi @spike04  @lordgreg 
I just deleted the existing flutter setup,  including pub cache and tool history, downloaded fresh zip for `2.0.1`, enabled macOS support and create a project and run the app, no issues, app launches  with `flutter -d run macos` 

<details> 
<summary>flutter doctor -v</summary> 

```bash
tahatesser@Tahas-MacBook-Pro foo % flutterp doctor -v
[✓] Flutter (Channel stable, 2.0.1, on macOS 11.2.2 20D80 darwin-x64, locale en-GB)
    • Flutter version 2.0.1 at /Users/tahatesser/Code/flutter
    • Framework revision c5a4b4029c (22 hours ago), 2021-03-04 09:47:48 -0800
    • Engine revision 40441def69
    • Dart version 2.12.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (3 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS 14.4
    • macOS (desktop)        • macos                     • darwin-x64     • macOS 11.2.2 20D80 darwin-x64
    • Chrome (web)           • chrome                    • web-javascript • Google Chrome 88.0.4324.192

• No issues found!
tahatesser@Tahas-MacBook-Pro foo % 

```

</details>

labelling for further triage
Thank you

**Comment by stuartmorgan** [comment]
> Adding the `-v` part where the exception occurs:

The exception just means that the Xcode build failed; all Xcode build failures no matter the cause will have that exception at the end. The actual build error is likely earlier in the log than what you've posted.

**Comment by jmagman** [comment]
```
... build/macos/Build/Products/Debug/prc_demo.app/Contents/Frameworks/FlutterMacOS.framework/FlutterMacOS: bundle format is ambiguous (could be app or framework)
```
This is a duplicate of https://github.com/flutter/flutter/issues/77173 and is due to a malformed framework on the 2.0.0 SDK packed on the website.  You shouldn't see this if you `flutter upgrade`d.  It's been fixed in stable on 2.0.1, beta, and in master.  The `dev` channel will get the fix on the next release.

The workaround is:
```
$rm -rf path/to/flutter/bin/cache/artifacts/engine/darwin-x64*
```
Then try building your macOS app again.

**Comment by lordgreg** [comment]
Thank you  @jmagman. Sadly, even after upgrading to 2.0.1 (see my flutter doctor above), the Problem persisted. I'll try with removing the engine cached. Thank you for your support.

**Comment by lordgreg** [comment]
Hi again @jmagman,

The solution you've posted didn't fixed the issue. I've tried the same use-case several times. After using the
```
flutter clean 
and
flutter precache -f
```

It went through and created the macos build.

**Comment by jmagman** [comment]
> Sadly, even after upgrading to 2.0.1 (see my flutter doctor above), the Problem persisted.

@lordgreg The packaging problem has been fixed in 2.0.1 meaning users who download 2.0.1 will not hit this problem.  If you are already in this state because you downloaded 2.0.0 you need run `rm -rf path/to/flutter/bin/cache/artifacts/engine/darwin-x64*`.  Upgrading to 2.0.1 won't fix it.

> The solution you've posted didn't fixed the issue.

You're saying removing `path/to/flutter/bin/cache/artifacts/engine/darwin-x64*` and rebuilding the macOS app didn't fix it?  That's kind of surprising since that's where the malformed frameworks are.
